src/app/clusters/general-diagnostics-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,141 @@
+// the static assert is skipped in case of dynamic endpoints.
+static_assert((GeneralDiagnostics::StaticApplicationConfig::kFixedClusterConfig.size() == 1 &&
+               GeneralDiagnostics::StaticApplicationConfig::kFixedClusterConfig[0].endpointNumber == kRootEndpointId) ||
+              GeneralDiagnostics::StaticApplicationConfig::kFixedClusterConfig.size() == 0);
+
+namespace {
+#if defined(ZCL_USING_TIME_SYNCHRONIZATION_CLUSTER_SERVER) || defined(GENERAL_DIAGNOSTICS_ENABLE_PAYLOAD_TEST_REQUEST_CMD)
+LazyRegisteredServerCluster<GeneralDiagnosticsClusterTimeSnapshotPayloadTestRequest> gServer;
+#else
+LazyRegisteredServerCluster<GeneralDiagnosticsCluster> gServer;
+#endif
+
+// compile-time evaluated method if "is <EP>::GeneralDiagnostics::<ATTR>" enabled
+constexpr bool IsAttributeEnabled(EndpointId endpointId, AttributeId attributeId)
+{
+    for (auto & config : GeneralDiagnostics::StaticApplicationConfig::kFixedClusterConfig)
+    {
+        if (config.endpointNumber != endpointId)
+        {
+            continue;
+        }
+        for (auto & attr : config.enabledAttributes)
+        {
+            if (attr == attributeId)
+            {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+} // namespace
+
+void emberAfGeneralDiagnosticsClusterInitCallback(EndpointId endpointId)
+{
+    VerifyOrDie(endpointId == kRootEndpointId);
+    const GeneralDiagnosticsEnabledAttributes enabledAttributes{
+        .enableTotalOperationalHours = IsAttributeEnabled(endpointId, Attributes::TotalOperationalHours::Id),
+        .enableBootReason            = IsAttributeEnabled(endpointId, Attributes::BootReason::Id),
+        .enableActiveHardwareFaults  = IsAttributeEnabled(endpointId, Attributes::ActiveHardwareFaults::Id),
+        .enableActiveRadioFaults     = IsAttributeEnabled(endpointId, Attributes::ActiveRadioFaults::Id),
+        .enableActiveNetworkFaults   = IsAttributeEnabled(endpointId, Attributes::ActiveNetworkFaults::Id),
+    };
+
+    gServer.Create(enabledAttributes);
+
+    CHIP_ERROR err = CodegenDataModelProvider::Instance().Registry().Register(gServer.Registration());
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(AppServer, "Failed to register GeneralDiagnostics on endpoint %u: %" CHIP_ERROR_FORMAT, endpointId,
+                     err.Format());
+    }
+}
+
+void emberAfGeneralDiagnosticsClusterShutdownCallback(EndpointId endpointId)
+{
+    VerifyOrReturn(endpointId == kRootEndpointId);
+    CHIP_ERROR err = CodegenDataModelProvider::Instance().Registry().Unregister(&gServer.Cluster());
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(AppServer, "Failed to unregister GeneralDiagnostics on endpoint %u: %" CHIP_ERROR_FORMAT, endpointId,
+                     err.Format());
+    }
+    gServer.Destroy();
+}
+
+void MatterGeneralDiagnosticsPluginServerInitCallback() {}
+
+void MatterGeneralDiagnosticsPluginServerShutdownCallback() {}
+
+namespace chip {
```

Comment:
supernit: c++17 would allow `namespace chip::app::Clusters::GeneralDiagnostics` which is shorter

https://en.cppreference.com/w/cpp/language/namespace.html (part 8 - nested namespaces)